packages/pwa/app/components/locale-selector-text/index.jsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {Text} from '@chakra-ui/layout'
+import React from 'react'
+import {FormattedMessage, defineMessages, useIntl, defineMessage} from 'react-intl'
+
+const LocaleSelectorText = ({localeId, asDropdownOption = false, ...otherProps}) => {
+    const found = LOCALE_MESSAGES[localeId]
+    const intl = useIntl()
+
+    if (found) {
+        return asDropdownOption ? (
+            <option value={localeId} {...otherProps}>
+                {intl.formatMessage(found)}
+            </option>
+        ) : (
+            <Text {...otherProps}>
+                <FormattedMessage {...found} />
+            </Text>
+        )
+    } else {
+        console.error(
+            `No locale message found for "${localeId}". Please update the list accordingly.`
+        )
+
+        return asDropdownOption ? (
+            <option value={localeId} {...otherProps}>
+                {intl.formatMessage(FALLBACK_MESSAGE, {localeId})}
+            </option>
+        ) : (
+            <Text {...otherProps}>
+                <FormattedMessage {...FALLBACK_MESSAGE} values={{localeId}} />
+            </Text>
+        )
```

Comment:
Gracefully handle the scenario of an unknown locale:
- Logging to console (and not throwing an error) so that the page would still render. This logging also helps to pinpoint which file needs to be updated.
- A fallback message for the unknown locale